### ISSUE TITLE
[Bullet] Duplicate body_shape_entered reported on CollisionShapes not close to the collision (even when disabled)

### ISSUE THREAD

**Issue by TZer0** [godotengine/godot#44644]
So I've searched through the issues, and I've seen some people complain about something similar, but I feel like I have boiled it down into the most fundamental here.

Please note that I'm new to Godot, so unless I'm doing something wrong in my body_shape_entered receiver in this particular line, then you can probably disregard this as a bug report (and instead view it as a documentation issue):
```
func _body_hit(body_id, body, body_shape, local_shape):
    shape_owner_get_owner(shape_find_owner(local_shape))
```

This also seems to apply to `body_shape_exited`.

**Godot version:**
3.2.3.stable.mono.official

**OS/device including version:**
Windows 10. Graphics not relevant.

**Issue description:**
It seems that sometimes two `body_shape_entered`-signals are emitted when a `RigidBody`has turned on contact monitor and set the contacts reported appropriately. One for the actual collision that occurred and sometimes one for the first `CollisionShape` in the `RigidBody`. This seems to occur more often with bodies moving quickly at the `RigidBody` in question.

This seems like an unintended effect as it is not consistent and can happen with `CollisionShape`s that are disabled.

**Steps to reproduce:**
1. Create a setup that involves a `RigidBody`with multiple `CollisionShape`s. Disable the first `CollisionShape`.
2. Create a spawner that fires a `RigidBody` over and over at this one (preferably ones that disappear after a second, with spawn rate set to one second).
3. Connect `body_shape_entered` and observe the passed values to that function.
4. Let it fire a couple of times, this bug is inconsistent.

**Minimal reproduction project:**
[bodyshapeenteredbug.zip](https://github.com/godotengine/godot/files/5737984/bodyshapeenteredbug.zip)
Sample output:
```
ALERT - DISABLED ENTRY:
[RigidBody:1812]
1812
0
0
0
@@2
disabled: True
--------

Correct entry:
[RigidBody:1812]
1812
0
5
5
@@12
disabled: False
--------
```
My demo adds a bunch of `CollisionShape`s and disables all but one, hence colliding with two different shapes should be impossible, yet this shows we collided with `@@12` (expected) and `@@2` (should be impossible as it is disabled). If you go into `Spawner.gd` and change `-100` to `0` on line 20, this stops happening.

**Comment by snaar** [comment]
Can you check if it's happening in 3.2.2? If not, then it's maybe it's caused by same change that did #44632 for me.

**Comment by snaar** [comment]
Ok it's unrelated, just tried and your issue reproduces in 3.2.2 too.

**Comment by snaar** [comment]
Poked a bit more since I was curious, it's already broken in 3.2 alpha0, but it's fine in 3.1. Unfortunately the build is broken a lot there so it's hard to bisect, but the issue is not there as of commit ee2c26a7b3, which is there 3.2 version starts according to the changelog, so it got broken somewhere between ee2c26a7b3 and 3.2 alpha0.

**Comment by snaar** [comment]
I did bisect, this seems to be the commit that causes your repro project to go bad: https://github.com/godotengine/godot/commit/c1c85555650aa4f976bd880a9b6ab7f21ba0f63a

**Comment by snaar** [comment]
Yeah, confirming that backing out that commit at the head of 3.2 branch does make the error message go away.

**Comment by madmiraal** [comment]
As identified by @snaar, this is related to CCD, which is enabled by default even when it shouldn't be (fixed with #43801, but not backported to 3.2). This is also related to #39767, i.e. reporting the wrong local collision shape, which is fixed by #42881, but hasn't been merged yet. However, this issue is not fixed with #42881, because this happens when Bullet CCD detects a collision with a `RigidBody` with multiple `CollisionShapes`. When a CCD collision is detected, it creates a collision manifold and an estimated contact point, which doesn't contain the shape data. When there is no shape data, Godot uses the first shape. This is the incorrect signal being emitted.
